client/src/partials/donors/donors.js

```diff
@@ -0,0 +1,160 @@
+angular.module('bhima.controllers')
+.controller('DonorsController', DonorsController);
+
+DonorsController.$inject = ['$http', '$translate', 'DonorService', 'SessionService', 'uuid'];
+
+/**
+ * Donors Controller
+ * This controller is about donors management module in the admin zone
+ * It's responsible for crud operations for donors
+ */
+function DonorsController($http, $translate, DonorService, Session, uuid) {
+  'use strict';
+
+  var vm = this;
+
+  /** breadcrumb configurations */
+  vm.bcPaths = [
+    {
+      label: $translate.instant('DONOR_MANAGEMENT.TITLE'),
+      current: true
+    }
+  ];
+  vm.bcButtons = [
+    {
+      dataMethod : 'create',
+      color  : 'btn-default',
+      icon   : 'glyphicon glyphicon-plus-sign',
+      label  : $translate.instant('DONOR_MANAGEMENT.NEW'),
+      action : create
+    }
+  ];
+
+  /** init variables */
+  vm.state = {};
+  vm.donor = {};
+
+  /** Expose to the view */
+  vm.update = update;
+  vm.cancel = cancel;
+  vm.submit = submit;
+  vm.remove = remove;
+  vm.refreshValidation = refreshValidation;
+
+  /** Load necessary data */
+  donorList();
+
+  function donorList() {
+    return DonorService.read()
+    .then(function (list) {
+      vm.donorList = list;
+    })
+    .catch(handler);
+  }
+
+  function update(id) {
+    initialise('action', 'update', id);
+  }
+
+  function create() {
+    initialise('action', 'create');
+  }
+
+  function remove(id) {
+    initialise('action', 'remove', id);
+  }
+
+  function cancel() {
+    initialise('default');
+  }
+
+  function submit(invalid) {
```

Comment:
Receiving invalid is confusing for me to read here, I would recommend passing in the form with a meaningful name - this provides context for _what_ is invalid. 

For example : 
```js
function submit(donorForm) { 
  if (donorForm.$invalid) { 
    return;
  }
  // else...
```

If it just remain this way a comment would be useful explaining what is invalid.